### ISSUE TITLE
TransferManager.doDownload() throws NPE when there is no Last-Modified header

### ISSUE THREAD

**Issue by artembilan** [aws/aws-sdk-java#822]
The code looks like:
````java
// Used to check if the object is modified between pause and resume
long lastModifiedTime = objectMetadata.getLastModified().getTime();
````
where:
````java
 /**
     * Gets the value of the Last-Modified header, indicating the date
     * and time at which Amazon S3 last recorded a modification to the
     * associated object.
     *
     * @return The date and time at which Amazon S3 last recorded a modification
     *         to the associated object. Returns <code>null</code> if
     *         the Last-Modified header hasn't been set.
     */
    public Date getLastModified() {
        return cloneDate((Date)metadata.get(Headers.LAST_MODIFIED));
    }
````
Pay attention to the note in the JavaDoc:
> Returns `<code>null</code>` if the Last-Modified header hasn't been set.

Therefore the in the `TransferManager` should be protected against NPE and maybe just returns `0` for the `lastModifiedTime` variable.

**Comment by varunnvs92** [comment]
How did you come across this issue? Can you give me a case where LAST_MODIFIED is null.
I was wondering if this may be a documentation bug as [Amazon S3 ](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-metadata) docs points this field is the object creation date or the last modified date, whichever is the latest. So once an object is created, I don't see a way for this field to have a null value.

**Comment by artembilan** [comment]
Well, my case was enough naive with the Mock test:
```
given(amazonS3.getObjectMetadata(any(GetObjectMetadataRequest.class))).willReturn(new ObjectMetadata());
```

That worked for me, just because I used AWS SDK - 1.10.30. Since there the download logic has been changed.

Right, now I just provide `new Date()` for the `LastModified` property of the `ObjectMetadata`.

I won't mind if that will be fixed on the `ObjectMetadata` level. For example asserting that `Last-Modified` header can't be `null` and so on for JavaDocs.

Thanks
